crates/amaru-ledger/src/rules/transaction/disjoint_ref_inputs.rs

```diff
@@ -0,0 +1,36 @@
+use amaru_kernel::{TransactionBody, TransactionInput};
+
+use crate::rules::TransactionRuleViolation;
+
+pub struct NonDisjointRefInputs {
+    pub intersection: Vec<TransactionInput>,
+}
+
+impl Into<TransactionRuleViolation> for NonDisjointRefInputs {
```

Comment:
I attempted to implement the `From` trait here instead, but I ran into a strange `rustc` error I'm still working on debugging.

Reply:
Looks like worth looking into it. Please share the rustc error if it can be helpful.